### ISSUE TITLE
Improve entity classes and implement CRUD operations 

### ISSUE THREAD

**Issue by firasrg** [firasrg/car-service-rest-api#6]
We need to add CRUD operations for entities following the [MVC pattern](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller). 

This includes, reviewing the current entity classes. In addition, implement the following operations :
1. Create a new entity.
2. Read: Fetch entities or get entity by ID.
3. Update: Modify an existing entity.
4. Delete: Remove an entity by ID.

### Requirements
1. Validation: Use Spring Validators.
2. Logging: Include logs for key actions and errors.
3. API Responses: create a generic class (ex: `Response<Object>`). 
4. Ensure status codes match HTTP methods and [Restful specifications.](https://restfulapi.net/http-status-codes/)

### Notes : 
- Persistence repositories should be accessed only in Service classes.
- Controllers should use service classes. 
- Services can call other services.
- Follow RESTful best practices.

**Comment by Fernand0-jf** [comment]
That's great, man. I'm joining this work

**Comment by Fernand0-jf** [comment]
That's great. I'm going to participate in this work. I am a Computer Science student.

**Comment by firasrg** [comment]
Hello @Fernand0-jf and welcome. Feel free to clone the repo and try it

**Comment by Singh916** [comment]
Hi @firasrg  I'm also contributing on this one.

**Comment by firasrg** [comment]
Hello @Singh916 ! That's great 😃 . It's open to anyone. We can do multiple PRs

**Comment by Singh916** [comment]
Sure make sense.

**Comment by Joao-Darwin** [comment]
Hello @firasrg! I'm a brazilian Backend Developer and i would like contribute for your project! Will add hateoas on controllers?

**Comment by firasrg** [comment]
Hello @Joao-Darwin ! Thanks for the suggestion. [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS) is indeed a valuable approach, but considering we are still in the early stages of the project, I think we should focus on establishing the core CRUD functionality first. HATEOAS adds a layer of complexity that might not be necessary for now. We can always revisit and discuss adding it later once the basic structure is solid. Feel free to open a separate issue for it with enough details on how you think to do it.

**Comment by Singh916** [comment]
Hi @firasrg  

While debugging the code , getting below errors and breakpoints are also not hit . Can you help here 
I'm using IntelliJ.

2024-08-13T00:19:09.522+05:30  INFO 403048 --- [car-service-rest-api] [           main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
2024-08-13T00:19:09.822+05:30  INFO 403048 --- [car-service-rest-api] [           main] o.s.d.j.r.query.QueryEnhancerFactory     : Hibernate is in classpath; If applicable, HQL parser will be used.
2024-08-13T00:19:10.614+05:30  INFO 403048 --- [car-service-rest-api] [           main] c.frg.carservice.CarServiceApplication   : Started CarServiceApplication in 4.057 seconds (process running for 4.453)
2024-08-13T00:19:10.722+05:30  INFO 403048 --- [car-service-rest-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2024-08-13T00:19:10.734+05:30  INFO 403048 --- [car-service-rest-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2024-08-13T00:19:10.737+05:30  INFO 403048 --- [car-service-rest-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown completed.
Disconnected from the target VM, address: '127.0.0.1:55072', transport: 'socket'

**Comment by firasrg** [comment]
Hi @Singh916! I don't see any errors on the log you shared, how did you run the app on your local machine ? I recommend to follow the official guide so you get it within the the right way. Otherwise, you may need to open a separate issue and provide full details about your environment (os, java version, IDE), logs and any changes u made.

**Comment by Singh916** [comment]
Yes I'm running on my local machine and just trying to debug the flow through CarController layer with breakpoint and debug mode in IntelliJ.

**Comment by jgraneros** [comment]
Hello!

Is this issue still open for contribution? I'd like to work on it.

**Comment by firasrg** [comment]
Hi @jgraneros ! and thank you for your interest. the project has been on hold for a while due to time constraints, but your message has reignited my motivation to revive it 🤩. Feel free to work on it in the way that suits you best. Once you submit a PR, we can review it together
